libcxx/include/__type_traits/is_core_convertible.h

```diff
@@ -30,6 +30,13 @@ template <class _Tp, class _Up>
 struct __is_core_convertible<_Tp, _Up, decltype(static_cast<void (*)(_Up)>(0)(static_cast<_Tp (*)()>(0)()))>
     : true_type {};
 
+#if _LIBCPP_STD_VER >= 17
+
+template <class _Tp>
```

Comment:
Why not just use `__is_core_convertible` directly? The additional instantiations might not be needed.

Reply:
I think it's better to introduce `__is_core_convertible_v` first and implement `__is_core_convertible` with it, then use `__is_core_convertible_v<decltype(...), bool>` in the constraints.

Reply:
@frederick-vs-ja I implemented `__is_core_convertible_v` for C++17 onward  similar to other traits. I don't understand why you suggest to do it in reverse? Everything seems fine as it is right now. Also we have the concept `__core_convertible_to`, do we want to keep it or replace it with `__is_core_convertible_v` for insignificant code reduction (of redundancy)?